### PR TITLE
Support timestamp

### PR DESCRIPTION
**Pull request by erikn69**
Timestamp

### CODE REVIEW

**Comment by dealfonso** [comment]
Hi,
what means "starting point"? Is it fully working or is it a work-in-progress?

**Comment by erikn69** [comment]
Fully working, but LTV is pending

Also, the code is from https://github.com/dealfonso/sapp/issues/8#issuecomment-1059107230 and https://github.com/dealfonso/sapp/issues/8#issuecomment-1470039444, 
and there are probably shorter and better ways to do it, without needing to add [phpseclib](https://github.com/phpseclib/phpseclib)

**Review by PaolaRuby — APPROVED** [review]

**Comment by hidasw** [comment]
@erikn69 


> Fully working, but LTV is pending
> 
> Also, the code is from [#8 (comment)](https://github.com/dealfonso/sapp/issues/8#issuecomment-1059107230) and [#8 (comment)](https://github.com/dealfonso/sapp/issues/8#issuecomment-1470039444), and there are probably shorter and better ways to do it, without needing to add [phpseclib](https://github.com/phpseclib/phpseclib)
> 
> Also i found this: `https://github.com/tecnickcom/TCPDF/pull/617`

It's great.
You right that its better if use self asn1 parser instead of [phpseclib](https://github.com/phpseclib/phpseclib).
I make simple code to parse and create asn1 [https://github.com/hidasw/PhpASN1](https://github.com/hidasw/PhpASN1) , maybe you can improve that.

File: src/PDFSignatureObject.php
Comment (hidasw) [inline src/PDFSignatureObject.php:1]: Can it be minimized by performing the signing process twice?
first signing with wide enough __SIGNATURE_MAX_LENGTH, ignore result. Just to calculate the size of signature data.
And last time is real signing with change __SIGNATURE_MAX_LENGTH based previously calculated signature data size.
I haven't tried it yet, but do you think so?

**Comment by erikn69** [comment]
>You right that its better if use self asn1 parser instead of [phpseclib](https://github.com/phpseclib/phpseclib).

I think the same, but I have no experience, I just copied the code from the proposed examples

File: src/PDFSignatureObject.php
Comment (erikn69) [reply src/PDFSignatureObject.php:1]: Yes, that would reduce the final size of the pdf, but couldn't it be a big change in performance?

Comment (hidasw) [reply src/PDFSignatureObject.php:1]: umm...that's right.
everywhere, efficiency and effectivity are always opposites.

**Comment by parallels999** [comment]
>You right that its better if use self asn1 parser instead of [phpseclib](https://github.com/phpseclib/phpseclib).
I make simple code to parse and create asn1 https://github.com/hidasw/PhpASN1 , maybe you can improve that.

that would be great

**Pull request closed without merging**